ffcx/ir/analysis/modified_terminals.py

```diff
@@ -168,7 +173,7 @@ def strip_modified_terminal(v):
     return v
 
 
-def analyse_modified_terminal(expr):
+def analyse_modified_terminal(expr, cell=None):
```

Comment:
Maybe we should call this `coordinate_cell`? I.e. the cell that the `coordinate-element` is using? 
I guess the doc-string should reflect this if we decide not to change the name.

Reply:
Since we are at it, I think we should add type-hints here, i.e.

```python

from ufl.classes import (
...
    Cell,
...
)
from ufl.core.expr import Expr

def analyse_modified_terminal(expr: Expr, cell: typing.Optional[Cell]=None):
    ....
```